controller/units.go

```diff
@@ -8,60 +8,61 @@ import (
 	"sync"
 
 	"github.com/elastic/elastic-agent-client/v7/pkg/client"
-	"github.com/elastic/elastic-agent-shipper/config"
 )
 
 // UnitMap is a wrapper for safely handling the map of V2 client units
 type UnitMap struct {
 	mut sync.Mutex
 	// The map of input-type units
-	// These will presumably be used during the startup phase of the shipper
-	// communicating with an input, but not sure how yet
-	inputUnits map[unitKey]ShipperUnit
-	outputUnit *client.Unit
-}
-
-// a unit ID isn't actually unique, the hash key must be a combination ID/type
-type unitKey struct {
-	ID   string
-	Type client.UnitType
+	inputUnits map[string]ShipperUnit
+	outputUnit ShipperUnit
 }
 
 // ShipperUnit wraps the available config for a unit
 type ShipperUnit struct {
 	Unit *client.Unit
-	// for now, we're treating the connection config for the unit a bit differently,
-	// as it appears that the connection is the same across the lifetime of a unit
-	Conn config.ShipperClientConfig
+	// When we get an updated unit, we don't actually get a "new" unit event, just a pointer to the same unit, which is updated under the hood
+	// which means anything in the parent unitMap hashmap is also updated.
+	// if we want to compare configs, log levels, etc, we need to save it off
```

Comment:
```go
// if we want to compare configs, log levels, etc, we need to save it off
```

Save it off what?

Reply:
The config from the unit, will add to that comment a bit...